crypto/pkcs7/pkcs7_test.cc

```diff
@@ -1830,15 +1828,15 @@ TEST(PKCS7Test, TestEnveloped) {
   // expectation. Ideally we'd find a way to access the padded plaintext and
   // account for this deterministically by checking the random "padding" and
   // adusting accordingly.
-  int max_decrypt =
-      sizeof(decrypted) + EVP_CIPHER_block_size(EVP_aes_128_cbc());
-  int decrypted_len = BIO_read(bio.get(), decrypted, max_decrypt);
-  if (decrypted_len > (int)pt_len) {
+  const size_t max_decrypt =
+    pt_len + EVP_CIPHER_block_size(EVP_aes_128_cbc());
+  const size_t decrypted_len = (size_t)BIO_read(bio.get(), decrypted, sizeof(decrypted));
```

Comment:
Might be a little bit more complicated, because `BIO_read` can return negative failure codes.

Reply:
Ok, I'll add a check to ensure that the value is in the expected range:
```
ASSERT_LE(decrypted_len, sizeof(decrypted));
```

Reply:
Ok with the note that `(size_t)BIO_read` converts signed to unsigned. So,
`ASSERT_LE(decrypted_len, sizeof(decrypted));` could be a bit awkward.